payment/payment.go

```diff
@@ -30,6 +30,13 @@ func FromPublicKey(pubkey *btcec.PublicKey, network *network.Network) Payment {
 	return Payment{network, pubkey, hash, nil, nil, nil}
 }
 
+// FromPayment creates a Payment struct from a another Payment
+func FromPayment(payment *Payment, network *network.Network) Payment {
```

Comment:
The network should be retrieved from the incoming payment, instead of being provided as parameter.